tests/rptest/tests/compatibility/sarama_test.py

```diff
@@ -11,13 +11,29 @@
 from ducktape.mark.resource import cluster
 from ducktape.utils.util import wait_until
 
-from rptest.services.kaf_producer import KafProducer
+from rptest.services.rpk_producer import RpkProducer
 from rptest.services.compatibility.example_runner import ExampleRunner
 import rptest.services.compatibility.sarama_examples as SaramaExamples
 from rptest.tests.redpanda_test import RedpandaTest
 from rptest.clients.types import TopicSpec
 
 
+class StringProducer(RpkProducer):
+    def __init__(self,
+                 context,
+                 redpanda,
+                 topic,
+                 msg_size,
+                 msg_count,
+                 acks=None):
+        super(StringProducer, self).__init__(context, redpanda, topic,
+                                             msg_size, msg_count, acks)
+
+    def value_gen(self):
+        msg_cmd = f'head -c1 /dev/urandom | md5sum | head -c{self._msg_size}'
```

Comment:
It's not obvious why we're doing this for loop and md5sum -- what was wrong with the existing RpkProducer behavior of tasking the urandom stream directly and chunking it up inside rpk?

Reply:
The original generator produced random bytes but the mechanism that ducky uses to read from the containers (it's called paramiko) had trouble decoding some of the bytes as `utf-8`.

The commit message for [https://github.com/vectorizedio/redpanda/pull/3483/commits/5da2d7dae9ac3534a577cd0f48e3f8d2c3a2b780](5da2d7d) explains that  alpha-numeric characters are needed instead. I can update the commit (I need to fix typos anyways).

The `md5sum` isn't necessary. We can use any other method to generate random characters.

Reply:
The [RpkConsumer](https://github.com/vectorizedio/redpanda/blob/dev/tests/rptest/services/rpk_consumer.py#L92) and [ExampleRunner](https://github.com/vectorizedio/redpanda/blob/dev/tests/rptest/services/compatibility/example_runner.py#L37) both use the same ducky method to read from containers.

So to clarify, it seems the Sarama consumer used in the test has trouble with the bytes generated from the original RpkProducer and that manifests as a `UnicodeDecodeError` in ducky.

I'll edit the commit message to be more clear and add in some comments to the code.

Reply:
Ah - in that case something that just prints human readable encodings is probably a better choice than md5sum to get printable ASCII bytes from urandom.  md5sum is pretty expensive, even on modern CPUs.

Here's a trick, using obscure hexdump format strings (add hexdump to the docker image): 
`dd if=/dev/urandom bs=1k count=1 | hexdump -e "1/1 \"%02x\""`

That'll get you 2048 bytes of human readable random bytes.  It's a deterministic length, so you can still figure out how many bytes you want, and then pipe the output into rpk, instead of having to use a bash for loop.

Could add a "printable" flag to RpkProducer that causes it to just divide the message size by two, and insert the extra hexdump step.

I'm being a bit picky here, but lots of these producer helpers have ended up biting us down the line for one reason or another, so would be good to make this one clean+efficient.